lib/modules/datasource/rpm/index.ts

```diff
@@ -0,0 +1,155 @@
+    key: ({ registryUrl }: GetReleasesConfig) => registryUrl ?? '',
+    ttlMinutes: 1440,
+  })
+  async getFilelistsXmlUrl(registryUrl: string): Promise<string | null> {
+    const repomdUrl = joinUrlParts(
+      registryUrl,
+      RpmDatasource.repomdXmlFileName,
+    );
+    const response = await this.http.getText(repomdUrl.toString());
+    if (response.statusCode !== 200) {
+      throw new Error(
+        `Failed to fetch repomd.xml from ${repomdUrl} (${response.statusCode})`,
+      );
+    }
+
+    // parse repomd.xml using XmlDocument
+    const data = new XmlDocument(response.body);
+    const dataElements = data.childrenNamed('data');
+
+    for (const dataElement of dataElements) {
+      if (dataElement.attr.type === 'filelists') {
+        const locationElement = dataElement.childNamed('location');
+        if (!locationElement) {
+          throw new Error(`No location element found in filelists.xml`);
+        }
+        const href = locationElement.attr.href;
+        if (!href) {
+          throw new Error(`No href found in filelists.xml`);
+        }
+        // replace trailing "repodata/" from registryUrl, if it exists, with a "/" because href includes "repodata/"
+        const registryUrlWithoutRepodata = registryUrl.replace(
+          /\/repodata\/?$/,
+          '/',
+        );
+        return joinUrlParts(registryUrlWithoutRepodata, href);
+      }
+    }
+
+    return null;
+  }
+
+  async getReleasesByPackageName(
+    filelistsXmlUrl: string,
+    packageName: string,
+  ): Promise<ReleaseResult | null> {
+    const response = await this.http.getText(filelistsXmlUrl);
+    if (response.statusCode !== 200) {
+      throw new Error(
+        `Failed to fetch filelists.xml from ${filelistsXmlUrl} (${response.statusCode})`,
+      );
+    }
+    // parse filelists.xml
+    const data = new XmlDocument(response.body);
+    const packages = data.childrenNamed('package');
+    if (!packages) {
+      throw new Error(`No packages found in filelists.xml`);
+    }
+    const releases: Release[] = [];
+    for (const pkg of packages) {
+      const name = pkg.attr.name;
+      if (name !== packageName) {
+        continue;
+      }
+      const version = pkg.childNamed('version')?.attr?.ver ?? '';
+      const rel = pkg.childNamed('version')?.attr?.rel;
+      let versionWithRel = version;
+      if (rel) {
+        // if rel is present, we need to append it to the version. Otherwise, ignore it.
+        // e.g. 1.0.0-1, 1.0.0-2, 1.0.0-3 or 1.0.0
+        versionWithRel += `-${rel}`;
+      }
+
+      // check if the versionWithRel isn't already in the releases
+      // One versionWithRel could have multiple revisions or releases
+      // e.g. 1.0.0-1, 1.0.0-2, 1.0.0-3
+      const existingRelease = releases.find(
+        (release) => release.version === versionWithRel,
+      );
+      if (!existingRelease) {
+        releases.push({
+          version: versionWithRel,
+        });
+      }
+    }
```

Comment:
What do you think about doing this with a `Map` instead of an `Array`. Define `releases` as `new Map<string, Release>()`. Then rewrite this as something like:

```ts
if (!releases.has(versionWithRel)) {
  releases.set(versionWithRel, { version: versionWithRel });
}
```

Reply:
oh I thought `getReleases` must return `releases: Release[]`? No?

I am up to anything more suitable.
In this case, do you mean storing as a map like
```
   [ "1.0.0": [1.0.0-1, 1.0.0-2],
     "1.0.1": [1.0.1-1]              ]
```

Reply:
Yeah, but you can do something like

```ts
return {
  releases: releases.keys(),
};
```

Reply:
I made the comment more because a `Map` is a more efficient data structure than an array for your use case. `Array#find` is `O(n)`, whereas `Map#has` is `O(1)`.